### DISCUSSION TITLE
Issue with Certificate Rule Selection in Course Edit

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#690] [General Discussion]
Hello Team,

In release 5.0, there is an issue with the certificate edit functionality. When adding a certificate to a course, there are two options for selecting certificate rules: "ALL" and "My Org". The problem arises when selecting "My Org" and then attempting to edit the added certificate rule for the same course. Despite selecting "My Org" initially, upon editing, different rules such as "My state teacher" are being displayed. This behaviour is inconsistent and unexpected.

Steps to Reproduce:
1. Navigate to the certificate add section in a course.
2. Add a certificate and select "My Org" as the certificate rule.
3. Save the changes.
4. Revisit the certificate edit section for the same course.
5. Attempt to edit the added certificate rule.

Expected Behavior:
Upon editing the certificate rule, the previously selected option "My Org" should be displayed consistently.

Actual Behavior:
Different certificate rules, such as "My state teacher", are displayed instead of the initially selected "My Org" option.

Additional Information:
Upon investigating the TenantPreferencesRead API for certificate rules, the expected options are "All" and "My Org User". However, the behavior encountered suggests that other rules, such as "My state teacher", are being applied incorrectly.

Screenshots:
Attach screenshots demonstrating the issue during both the initial selection and subsequent editing of certificate rules.
![image (7)](https://github.com/Sunbird-Ed/Community/assets/72926048/587f9297-f7aa-4cc4-a9df-38b41a2e7cab)
![image (6)](https://github.com/Sunbird-Ed/Community/assets/72926048/d06ce72a-a256-4767-83a9-ab2dd49b1ff2)

Further Investigation:
Explore why the "My state teacher" rule is being applied instead of the expected "My Org" rule during certificate editing for courses. Determine if this issue is widespread or specific to certain configurations. Evaluate possible solutions to ensure consistent and accurate certificate rule selection behavior.

**Comment by ashokreddy1208** [comment]
@AmiableAnil @Hari-stackroute Request you to take a look into this query.

Note:- In Sunbird 'All', 'My state teacher' values are enabled, not "My Org", looks like new value is added to certificate rules which is working while adding new certificate but not reflecting in edit certificate screen.

**Comment by anisha147** [comment]
Hi @Hari-stackroute @AmiableAnil: Request you to look into the query.


cc: @ashokreddy1208

**Reply by AmiableAnil** [reply]
@siddhishinde0723 Can you please schedule a call for tomorrow for the same. Also if you can share me the API detail where you are facing this issue.

**Reply by siddhishinde0723** [reply]
@AmiableAnil  Sure I'll schedule a call for tomorrow, and there is no API, which is failing, and I don't think this is from the API.

**Reply by siddhishinde0723** [reply]
@AmiableAnil  Could you please let me know what time works best for you so we can schedule a meeting?

**Reply by AmiableAnil** [reply]
@siddhishinde0723 can you please schedule a call on 8th April at 11:00AM?

**Comment by HarishGangula** [comment]
As we there are no active updates, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
